### PR TITLE
doc(serverless/aws-lambda)/Add documentation for captureAllSettledReasons option

### PR DESCRIPTION
**Pull request by J4YF7O**
Follow https://github.com/getsentry/sentry-javascript/pull/4130

### CODE REVIEW

File: src/platforms/node/guides/aws-lambda/index.mdx [hunk 1]
```diff
@@ -80,6 +80,36 @@ exports.handler = Sentry.AWSLambda.wrapHandler(yourHandler, {
 });
 ```
 
+
+## Capture all errors when `Promise.allSettled` is returned by the handler
```
Comment (imatwawana) [inline src/platforms/node/guides/aws-lambda/index.mdx:84]:
```suggestion
## `Promise.allSettled` Returned by the Handler
```

File: src/platforms/node/guides/aws-lambda/index.mdx [hunk 2]
```diff
@@ -80,6 +80,36 @@ exports.handler = Sentry.AWSLambda.wrapHandler(yourHandler, {
...
+_(New in version ??)_
+
+By default, sentry capture the error raised by your handler.
```
Comment (imatwawana) [inline src/platforms/node/guides/aws-lambda/index.mdx:87]:
```suggestion
By default, Sentry captures errors raised by your handler.
```

File: src/platforms/node/guides/aws-lambda/index.mdx [hunk 3]
```diff
@@ -80,6 +80,36 @@ exports.handler = Sentry.AWSLambda.wrapHandler(yourHandler, {
...
+It's possible that your handler returns a `Promise.allSettled` result.
```
Comment (imatwawana) [inline src/platforms/node/guides/aws-lambda/index.mdx:88]:
```suggestion
However, your handler might return a `Promise.allSettled` result.
```

File: src/platforms/node/guides/aws-lambda/index.mdx [hunk 4]
```diff
@@ -80,6 +80,36 @@ exports.handler = Sentry.AWSLambda.wrapHandler(yourHandler, {
...
+In this case even if one of the messages has failed, no exception will be captured by Sentry.
```
Comment (imatwawana) [inline src/platforms/node/guides/aws-lambda/index.mdx:89]:
```suggestion
In this case, even if one of the messages has failed, Sentry won't capture any exception.
```

File: src/platforms/node/guides/aws-lambda/index.mdx [hunk 5]
```diff
@@ -80,6 +80,36 @@ exports.handler = Sentry.AWSLambda.wrapHandler(yourHandler, {
...
+
+The `captureAllSettledReasons` (default: `false`) option capture all promise rejected results.
```
Comment (imatwawana) [inline src/platforms/node/guides/aws-lambda/index.mdx:91]:
```suggestion
The `captureAllSettledReasons` (default: `false`) option captures all promise rejected results:
```

**Review by imatwawana — CHANGES_REQUESTED** [review]
Thanks for submitting this! This is just a review of the language and technical review will happen next.

**Comment by J4YF7O** [comment]
Sorry for my french... Thanks for the corrections :) 

Moreover I have no idea about the version, how will you manage this case ? 🤔

**Review by imatwawana — APPROVED** [review]

File: src/platforms/node/guides/aws-lambda/index.mdx [hunk 6]
```diff
@@ -80,6 +80,36 @@ exports.handler = Sentry.AWSLambda.wrapHandler(yourHandler, {
 });
 ```
 
+
+## `Promise.allSettled` Returned by the Handler
+_(New in version ??)_
```
Comment (iker-barriocanal) [inline src/platforms/node/guides/aws-lambda/index.mdx:85]: If there's no version set, we shouldn't add this line.
```suggestion
```

File: src/platforms/node/guides/aws-lambda/index.mdx [hunk 7]
```diff
@@ -80,6 +80,36 @@ exports.handler = Sentry.AWSLambda.wrapHandler(yourHandler, {
...
+
+```javascript {tabTitle:promiseSettledDefault}
+exports.handler = Sentry.AWSLambda.wrapHandler(() => {
+  return Promise.allSettled([
+        Promise.rejected(new Error('first')),
+        Promise.rejected(new Error('second'))
+  ]);
+});
+// Nothing captured
+```
```
Comment (iker-barriocanal) [inline src/platforms/node/guides/aws-lambda/index.mdx:101]: This is the default behavior, I don't think adding this code snippet provides much value.

File: src/platforms/node/guides/aws-lambda/index.mdx [hunk 8]
```diff
@@ -80,6 +80,36 @@ exports.handler = Sentry.AWSLambda.wrapHandler(yourHandler, {
...
+
+```javascript {tabTitle:captureAllSettledReasons}
+exports.handler = Sentry.AWSLambda.wrapHandler(() => {
+  return Promise.allSettled([
+        Promise.rejected(new Error('first')),
+        Promise.rejected(new Error('second'))
```
Comment (iker-barriocanal) [inline src/platforms/node/guides/aws-lambda/index.mdx:107]: In order to be consistent with the code style in the docs:
```suggestion
        Promise.rejected(new Error('second')),
```

File: src/platforms/node/guides/aws-lambda/index.mdx [hunk 9]
```diff
@@ -80,6 +80,36 @@ exports.handler = Sentry.AWSLambda.wrapHandler(yourHandler, {
...
+  ]);
+}, {captureAllSettledReasons: true});
```
Comment (iker-barriocanal) [inline src/platforms/node/guides/aws-lambda/index.mdx:109]: Same as above, to be consistent with the code style in the docs:
```suggestion
}, { captureAllSettledReasons: true });
```

File: src/platforms/node/guides/aws-lambda/index.mdx [hunk 10]
```diff
@@ -80,6 +80,36 @@ exports.handler = Sentry.AWSLambda.wrapHandler(yourHandler, {
 });
 ```
 
+
+## `Promise.allSettled` Returned by the Handler
```
Comment (iker-barriocanal) [inline src/platforms/node/guides/aws-lambda/index.mdx:84]:
```suggestion
## Capture Rejected Promises in `Promise.allSettled`
```

**Review by iker-barriocanal — CHANGES_REQUESTED** [review]
Thanks a lot for contributing! Some small changes to align with the docs style.

File: src/platforms/node/guides/aws-lambda/index.mdx (hunk 6 above)
Comment (J4YF7O) [reply src/platforms/node/guides/aws-lambda/index.mdx:85]: There will be a version, right? For the moment I don't know it, but the feature didn't exist before? It's better not to put anything in this case ? 

https://github.com/getsentry/sentry-docs/pull/4374#issuecomment-964273769

File: src/platforms/node/guides/aws-lambda/index.mdx (hunk 7 above)
Comment (J4YF7O) [reply src/platforms/node/guides/aws-lambda/index.mdx]: remove exemple
1e66df5

File: src/platforms/node/guides/aws-lambda/index.mdx (hunk 6 above)
Comment (AbhiPrasad) [reply src/platforms/node/guides/aws-lambda/index.mdx:85]: Yup it'll be with the next release, we can just update this PR when that goes out.

**Review by AbhiPrasad — APPROVED** [review]
Version should be `6.14.2` as per: https://github.com/getsentry/sentry-javascript/blob/master/CHANGELOG.md#6142

Thanks a lot for the SDK PR + docs, appreciate it!

**Comment by J4YF7O** [comment]
Version set in f8f45f5eed7bb86d5fd1880ea07b8969f10cdf7f

PR is ready to merge (IMO)

**Pull request merged**